### DISCUSSION TITLE
accessing a flag that doesn't exist

### DISCUSSION THREAD

**Discussion by till** [urfave/cli#1810] [Q&A] [answered]
I frequently run into an issue where I typo the name of a flag I defined.

So let's say, I defined a flag of name `connection` and later in code, I access it as `cCtx.String("connect")`. The latter returns an empty string and then usually leads to a cascading failure somewhere else. I also want to say: I do realize, this is a bug in my code.

Having said all this, I know I could wrap my code with another guard but that also seems like it'll lead to bloat.

I am wondering if anyone has any thoughts on this. E.g. I was wondering what people think about using `panic()` in the code, or setting a very obvious value of `**THIS DOES NOT EXIST**` or something like that in the library to make it more obvious where the problem is?

**Comment by dvaumoron** [comment, marked as answer]
The good practice is to use a const for your flag name.

Edit: You can also use the Destination field of StringFlag struct (if you don't bother to allocate the var when it's not needed).

**Reply by dearchap** [reply]
Also you can set the App.InvalidFlagAccessFunc to see if any invalid flag names are being accessed and do a panic there.
